front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx

```diff
@@ -67,41 +65,24 @@ const StdcmConsist = ({ isDebugMode, consistErrors = {}, disabled = false }: Std
     prefillConsist,
   } = useStdcmConsist();
 
-  const { filters, searchRollingStock, searchRollingStockById, filteredRollingStockList } =
-    useFilterRollingStock({ isStdcm: true });
+  const { filteredRollingStockList: rollingStocks } = useFilterRollingStock({ isStdcm: true });
 
-  const {
-    filteredTowedRollingStockList,
-    searchTowedRollingStock,
-    searchTowedRollingStockById,
-    filters: towedRsFilters,
-  } = useFilterTowedRollingStock({ isDebugMode });
-
-  useEffect(() => {
-    if (towedRollingStock) {
-      searchTowedRollingStock(towedRollingStock.name);
-    } else {
-      searchTowedRollingStock('');
-    }
-  }, [towedRollingStock]);
+  const { filteredTowedRollingStockList: towedRollingStocks } = useFilterTowedRollingStock({
+    isDebugMode,
+  });
 
   const getLabel = (rs: LightRollingStockWithLiveries) => {
     const secondPart = rs.metadata?.series || rs.metadata?.reference || '';
     return secondPart ? `${rs.name} - ${secondPart}` : rs.name;
   };
 
-  const onInputClick = () => {
-    if (rollingStock?.id !== undefined) {
-      searchRollingStockById(rollingStock.id);
-    }
-  };
-
-  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    searchRollingStock(e.target.value);
-    if (e.target.value.trim().length === 0) {
-      dispatch(updateRollingStockID(undefined));
-    }
-  };
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  const rollingStockComboBoxDefaultProps = useDefaultComboBox(rollingStocks, getLabel);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  const towedRollingStockComboBoxDefaultProps = useDefaultComboBox(
```

Comment:
Seems we don't need them
```suggestion
  const rollingStockComboBoxDefaultProps = useDefaultComboBox(rollingStocks, getLabel);
  const towedRollingStockComboBoxDefaultProps = useDefaultComboBox(
```

Reply:
I have errors in vsc if I remove them :/

Reply:
I can remove without error as well.